iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergObjectInspector.java

```diff
@@ -135,6 +135,9 @@ public ObjectInspector primitive(Type.PrimitiveType primitiveType) {
         return adjustToUTC ? TIMESTAMP_INSPECTOR_WITH_TZ : TIMESTAMP_INSPECTOR;
       case TIME:
         return IcebergTimeObjectInspector.get();
+      case TIMESTAMP_NANO:
+        boolean adjust = ((Types.TimestampNanoType) primitiveType).shouldAdjustToUTC();
+        return adjust ? TIMESTAMP_INSPECTOR_WITH_TZ : TIMESTAMP_INSPECTOR;
```

Comment:
```
case TIMESTAMP_NANO:
        boolean adjust = ((Types.TimestampNanoType) primitiveType).shouldAdjustToUTC();
        return adjust ? TIMESTAMP_INSPECTOR_WITH_TZ : TIMESTAMP_INSPECTOR;
```

I don't see any "nanos" usage in `IcebergTimestampWithZoneObjectInspectorHive3` i.e. `TIMESTAMPTZ_INSPECTOR_CLASS` , should we create a new class extending `AbstractPrimitiveJavaObjectInspector` for handling nano case? Also  `TIMESTAMP_INSPECTOR` will not return "nanosecond timestamp" https://github.com/apache/hive/blob/eaa8189600c74efa35cc9975eea0f81df59e5088/iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergTimestampObjectInspectorHive3.java#L41-L43

Reply:
AFAIK
Nanosecond precision is preserved because all conversions are done using:
* java.time.ZonedDateTime
* java.time.OffsetDateTime
* org.apache.hadoop.hive.common.type.TimestampTZ

All of these internally store time with nanosecond resolution

If you check the q file output, I have a case for all 4 types, and for the nanosecond timezones, it would be more digits

Reply:
Based on this I guess, new class for timestamp nano needs to be created. Maybe `IcebergTimestampNanoWithZoneObjectInspector and IcebergTimestampNanoObjectInspector`